File: testdata/csharp/clean.cs
using System;
using System.Security.Cryptography;

// Negative fixture: none of the ZS-CS rules should fire here.
public class CleanExample
{
    public void Greet(string visitorName)
    {
        var greeting = "hello, " + Sanitize(visitorName);
        Console.WriteLine(greeting);
    }

    public byte[] HashData(byte[] data)
    {
        var sha256 = SHA256.Create();
        return sha256.ComputeHash(data);
    }

    private string Sanitize(string value)
    {
        return value.Replace("<", "&lt;").Replace(">", "&gt;");
    }
}

File: testdata/java/clean.java
// Negative fixture: none of the ZS-JAVA rules should fire here.
import java.security.MessageDigest;

public class Clean {
    void run() throws Exception {
        String greeting = "hello";
        System.out.println(greeting);

        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        digest.update(greeting.getBytes());
    }
}
